### DISCUSSION TITLE
Package for NeoVim

### DISCUSSION THREAD

**Discussion by KoosieComputes** [cursorless-dev/cursorless#2065] [Ideas]
I'm imagining the combination of rapid vim movements (keyboard input) with a cursorless insert mode to be the ultimate speed combination. I have no idea what the development challenges would be to port cursorless over as a Neovim plugin

**Comment by auscompgeek** [comment]
Most folks that use both Cursorless and Neovim do so in VSCode. There's a really robust vscode-neovim extension.

**Comment by pokey** [comment]
We do plan to eventually port to other editors. For now, as @auscompgeek mentions, your best bet is to use [vscode-neovim](https://marketplace.visualstudio.com/items?itemName=asvetliakov.vscode-neovim), which runs a full-fledged headless neovim instance in the background. Fwiw I was a vim user for a decade, and use vscode-neovim by keyboard, but by voice I find that Cursorless subsumes everything for which I used to use neovim commands, and I just have it disabled when I'm voice coding

We are experimenting with a POC Cursorless keyboard mode, and for that I do use it in tandem with vscode neovim

**Comment by BlueDrink9** [comment]
For those people that do use both, is there a way to automatically enter insert mode on cursorless change commands etc?

**Reply by BlueDrink9** [reply]
This is working for now 
```talon

mode: command
mode: user.cursorless_spoken_form_test
tag: user.cursorless
-

# Override the cursorless change command to also enter insert mode.
change <user.cursorless_target>:
    user.cursorless_command("clearAndSetSelection", cursorless_target)
    user.run_rpc_command("vscode-neovim.lua", "vim.cmd('startinsert')")



```
